pkg/crd/known_types.go

```diff
@@ -55,7 +55,11 @@ var KnownPackages = map[string]PackageOverride{
 	"k8s.io/apimachinery/pkg/api/resource": func(p *Parser, pkg *loader.Package) {
 		p.Schemata[TypeIdent{Name: "Quantity", Package: pkg}] = apiext.JSONSchemaProps{
 			// TODO(directxman12): regexp validation for this (or get kube to support it as a format value)
-			Type: "string",
+			XIntOrString: true,
+			AnyOf: []apiext.JSONSchemaProps{
+				{Type: "integer"},
+				{Type: "string"},
```

Comment:
@sttts if we use intorstring, are we allowed to further restrict the variants and still be structural?  i.e.

```yaml
anyOf:
- type: int
- type: string
  pattern: "^(\\+|-)?(([0-9]+(\\.[0-9]*)?)|(\\.[0-9]+))(([KMGTIE]i)|[mkMGTPE]|((\\+|-)?(([0-9]+(\\.[0-9]*)?)|(\\.[0-9]+))))?$"

Reply:
(I think that matches the productions in `k8s.io/apimachinery/pkg/api/resource/quantity.go`)

Reply:
(also, we should really just have `format: quantity` supported in k/k)

Reply:
I like that idea about format. We should add that. For old servers though we will need the pattern.